ONSdigital/eq-author-app
# Ear 1995 view publish history

### What is the context of this PR?

Changes expand on the existing 'Publish' page, to add a table of publishing history

### Jira links

https://jira.ons.gov.uk/browse/EAR-1995

### How to review

If not already present, add the feature flag 'publishPage' to view the page

1. Create a new questionnaire, without duplicating.
2. Complete all required fields on the questionnaire so that no validation errors are present
3. Ensure the Publish option is visible in the left nav
4. Click the publish option to get the Publish page.
5. Check that the page displays the 'Publishing history' section after the 'Publish questionnaire' button.
6. Confirm that with no current history, the message 'No versions of this questionnaire have been published' is displayed
7. Click the 'Publish questionnaire' button
8. Confirm that the page refreshes to replace the 'no versions' message with a table.
9. Confirm the page has one data row, displaying the correct date/time, surveyID, formType, CIR Version and CIR ID.
10. Click the 'Publish questionnaire' button again, and confirm a new data row is added to the top of the table, with the latest date/time, and a different CIR ID. CIR version will still show '1'. This is expected, and will be resolved once actual CIR integration is complete.
11. More rows can be added if wanted by clicking the 'Publish questionnaire' button. After each click, confirm the newest entry is at the top of the table, with table rows ordered by the date/time, most recent first, and that the table updates automatically.
12. Return to the Author initial screen, and duplicate the questionnaire.
13. Open the duplicate questionnaire, and navigate to it's Publish page. 
14. Confirm that the duplicate questionnaire has no publish history, and does not have the publish history of the original
